langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/gemini/FunctionMapper.java

```diff
@@ -0,0 +1,79 @@
+package dev.langchain4j.model.gemini;
+
+import com.google.gson.Gson;
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolParameters;
+import dev.langchain4j.agent.tool.ToolSpecification;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+class FunctionMapper {
+
+    static GeminiTool fromToolSepcsToGTool(List<ToolSpecification> specifications) {
+        if (specifications == null) {
+            return null;
+        }
+
+        GeminiTool.GeminiToolBuilder tool = GeminiTool.builder();
+
+        List<GeminiFunctionDeclaration> functionDeclarations = specifications.stream()
+            .map(specification -> {
+                // using Gemini's built-in hosted Python sandboxed code executor
+                if (specification.name().equals("code_execution")) {
```

Comment:
I haven't touched this one.
I could use a more specific name instead of the name that Gemini uses?
Maybe `gemini_code_execution`?  (which is hopefully a name no user would use)

Reply:
Do I understand that `code_execution` enables sandbox code execution on Gemini side? If yes, perharps it makes more sense to have a flag in ctor like "enableCodeExecution" than expecting user to pass `ToolSpecification` with specific name? Also it does not seem right, because there is actually no tool on the client side, it is executed automatically on the LLM side